src/deepsparse/server/monitoring.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import multiprocessing
+import time
+from pathlib import Path
+from typing import List, Optional, Tuple
+
+import pydantic
+import requests
+import yaml
+
+from deepsparse.server.config import EndpointConfig, ServerConfig
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def start_file_watcher(
+    config_path: str, endpoints_url: str, check_interval_s: float
+) -> multiprocessing.Process:
+    """
+    Creates and starts a separate process that monitors `config_path`
+    for changes every `check_interval_s` seconds.
+    If endpoints are modified it sends requests to `endpoints_url`
+    to update the server.
+
+    :param config_path: Path to file to watch
+    :param endpoints_url: Url to send requests to for updating
+    :param check_interval_s: Seconds between each check.
+    :return: A **started** daemon `Process`.
+    """
+
+    proc = multiprocessing.Process(
+        target=_file_watcher,
+        kwargs=dict(
+            config_path=config_path,
+            endpoints_url=endpoints_url,
+            check_interval_s=check_interval_s,
+        ),
+        daemon=True,
+    )
+    proc.start()
+    return proc
+
+
+def _file_watcher(config_path: str, endpoints_url: str, check_interval_s: float):
```

Comment:
Unsure of how to test this, thoughts? Open to restructuring this function to make it easier to test. I think the `while True` currently makes it a bit hard to test, and there's no output from this function, so you can't black box it.

Reply:
could  test through `start_file_watcher` with the update method mocked as long as there's a way we can verify that the mocked method is called